definitions/npm/react-i18next_v1.10.x/flow_v0.36.x-/react-i18next_v1.10.x.js

```diff
@@ -0,0 +1,35 @@
+import type { TFunction } from 'i18next'
+
+declare module 'react-i18next' {
+  declare type TFunction = TFunction;
+  declare type Locales = string | Array<string>;
+
+  declare type StatelessComponent<P> = (props: P) => ?React$Element<any>;
+
+  declare type Comp<P> = StatelessComponent<P> | Class<React$Component<*, P, *>>;
+
+  declare type Translator<OP, P> = {
+    (component: StatelessComponent<P>): Class<React$Component<void, OP, void>>;
+    <Def, St>(component: Class<React$Component<Def, P, St>>): Class<React$Component<Def, OP, St>>;
```

Comment:
No, not the generics syntax in general, just this particular instance where `<Def, St>` comes before the function parameters.

I guess there's two questions:

1. Is this:

```js
type Foo<A> = <B>(bar: Bar<A, B>) => Bar<A, B>
```

Equivalent to this?

```js
type Foo<A> = {
  <B>(bar: Bar<A, B>): Bar<A, B>
}
```

2. In the above, is it true that `<B>` correspond to the type of the `bar` parameter?

```js
const foo: Foo<A> = blah
const bar: Bar<A, B> = blah
const baz = foo(bar) // baz has type Bar<A, B>
```

So my understanding would be that this syntax is useful when the type of the return value of a function depends on the type of the input value?